synapse/lib/trigger.py

```diff
@@ -182,6 +198,82 @@ async def runTagDel(self, node, tag, view=None):
                 for _, trig in globs.get(tag):
                     await trig.execute(node, vars=vars, view=view)
 
+    async def runEdgeAdd(self, n1, edge, n2, view=None):
+        n1form = n1.form.name
+        n2form = n2.form.name
+        vars = {'auto': {'opts': {'edge': edge, 'n2': n2}}}
+        with self._recursion_check():
+            # try all the variations of the optional params
```

Comment:
lots of work being done here. any ideas on how it effects perf? perhaps an overall strategy for caching all the triggers for a given n1+verb+n2 would be warranted?